src/core/lib/security/authorization/grpc_authorization_policy_provider.cc

```diff
@@ -50,21 +50,13 @@ StaticDataAuthorizationPolicyProvider::Create(absl::string_view authz_policy) {
 }
 
 StaticDataAuthorizationPolicyProvider::StaticDataAuthorizationPolicyProvider(
-    absl::InlinedVector<Rbac, 2> policies) {
-  GPR_ASSERT(policies.size() <= 2);
-  if (policies.size() == 1) {
-    GPR_ASSERT(policies[0].action == Rbac::Action::kAllow);
-    allow_engine_ =
-        MakeRefCounted<GrpcAuthorizationEngine>(std::move(policies[0]));
-  } else {
-    GPR_ASSERT(policies[0].action == Rbac::Action::kDeny);
-    GPR_ASSERT(policies[1].action == Rbac::Action::kAllow);
-    deny_engine_ =
-        MakeRefCounted<GrpcAuthorizationEngine>(std::move(policies[0]));
-    allow_engine_ =
-        MakeRefCounted<GrpcAuthorizationEngine>(std::move(policies[1]));
-  }
-}
+    RbacPolicies policies)
+    : allow_engine_(MakeRefCounted<GrpcAuthorizationEngine>(
+          std::move(policies.allow_policy))),
+      deny_engine_(policies.deny_policy.has_value()
+                       ? MakeRefCounted<GrpcAuthorizationEngine>(
+                             std::move(policies.deny_policy.value()))
```

Comment:
This can use `*policies.deny_policy` instead of `policies.deny_policy.value()`, since you've already checked `has_value()` above.

Reply:
Done